### PR TITLE
*: support `show stats_extended` to inspect extended stats cache

### PR DESCRIPTION
**Pull request by eurekaka**
### What problem does this PR solve?

Issue Number: close https://github.com/pingcap/tidb/issues/23033

Problem Summary:

Not a problem, provide a new SQL interface to inspect extended stats cache

### What is changed and how it works?

What's Changed:

Grab the extended stats cache and convert data to a readable format.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:

### Check List 

Tests 

- Unit test

Side effects

N/A

### Release note 

- Support `show stats_extended` to inspect extended stats cache

### CODE REVIEW

**Review by winoros — COMMENTED** [review]
/lgtm

File: executor/show_stats.go [hunk 1]
```diff
@@ -14,17 +14,88 @@
...
 )
 
+func (e *ShowExec) fetchShowStatsExtended() error {
+	do := domain.GetDomain(e.ctx)
+	h := do.StatsHandle()
+	dbs := do.InfoSchema().AllSchemas()
+	for _, db := range dbs {
+		for _, tblInfo := range db.Tables {
+			pi := tblInfo.GetPartitionInfo()
+			// Extended statistics for paritioned table is not supported now.
```
Comment (time-and-fate) [inline executor/show_stats.go:39]: paritioned => partitioned

**Review by time-and-fate — APPROVED** [review]
/lgtm

**Review by time-and-fate — APPROVED** [review]
lgtm

**Pull request merged**